realm/stats.gno

```diff
@@ -0,0 +1,130 @@
+package chess
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+var allPlayerStats avl.Tree // std.Address -> *playerStats
+
+type playerStats struct {
+	Addr          std.Address // Not stored when in avl.Tree, but lazily filled for public-facing helpers returning playerStats.
+	Moves         uint
+	StartedGames  uint
+	WonGames      uint
+	LostGames     uint
+	TimedoutGames uint
+	ResignedGames uint
+	DrawnGames    uint
+	SeriousGames  uint // finished, or resigned/drawn after 20 full moves (40 turns), used for the raffle.
+	// later we can add achievements:
+	// SuperFastAchievement // if a game is finished in less than N seconds.
+	// OnlyPawnsAchievement // winning with only pawns, etc.
+}
+
+func (s playerStats) String() string {
+	return ufmt.Sprintf(
+		"addr:%s moves:%d started:%d won:%d lost:%d timedout:%d resigned:%d drawn:%d serious:%d",
+		s.Addr, s.Moves, s.StartedGames, s.WonGames, s.LostGames, s.TimedoutGames,
+		s.ResignedGames, s.DrawnGames, s.SeriousGames,
+	)
+}
+
+func getPlayerStats(addr std.Address) *playerStats {
+	addrStr := string(addr)
+	res, found := allPlayerStats.Get(addrStr)
+	if found {
+		return res.(*playerStats)
+	}
+
+	newStats := playerStats{}
```

Comment:
Why not 
```go
newStats := playerStats{Addr: addr} 
```
?

And so why doing that in the public function instead (line 50) ?

Reply:
to save on storage since the address is already known by the `avl.Tree`; but I'm okay to move to your solution too. Feel free to patch my PR if you prefer 👍

Reply:
No that's fine, it's also good to save storage!